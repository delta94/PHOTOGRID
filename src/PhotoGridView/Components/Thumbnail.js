import React, { Component } from 'react';
import { PropTypes } from 'prop-types';

import { ImageOver } from '../Partials/ImageOver';
import { StatusButtonGroup } from '../Partials/StatusButtonGroup';

export class Thumbnail extends Component {
  constructor() {
    super();

    this.state = {
      isOver: false
    };
  }

  // toggleOver = () => {
  //   this.setState({
  //     isOver: !this.state.isOver
  //   },() => {
  //     console.log('isOver', this.state.isOver);
  //   });
  // };
  showOverlay = () => {
    this.setState({
      isOver: true
    });
  }
  hideOverlay = (e) => {
    if (e.target.width !== 0) {
      this.setState({
        isOver: false
      });
    }
  }

  render() {
    const { props } = this;
    return (
      <div
        style={{backgroundImage: `url(${props.photo.imageUrl})`}}
        className="photo-grid-thumbnail-container br__radius-5"
        onMouseEnter={this.showOverlay}
        onMouseLeave={(e) => this.hideOverlay(e)}
      >
        <StatusButtonGroup
          photo={props.photo}
        />
        {
          props.photo.mentioned &&
          <div className="triangle">
            <span className="icon__star triangle__star font-14"></span>
          </div>
        }
        {
          this.state.isOver &&
          <ImageOver
            photo = {props.photo}
            selectedPhotos={props.selectedPhotos}
            onStatusChange = {props.onStatusChange}
            onRightsRequest = {props.onRightsRequest}
            onStickyChange = {props.onStickyChange}
            onSelectionChange = {props.onSelectionChange}
          />
        }
      </div>
    );
  }
}

Thumbnail.propTypes = {
  photo: PropTypes.object,
  selectedPhotos: PropTypes.array,
  onStatusChange: PropTypes.func,
  onRightsRequest: PropTypes.func,
  onStickyChange: PropTypes.func,
  onSelectionChange: PropTypes.func
};

export default Thumbnail;