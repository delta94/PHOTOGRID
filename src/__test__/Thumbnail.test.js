import React from 'react';
import { shallow, mount } from 'enzyme';

import Thumbnail from '../PhotoGridView/Components/Thumbnail';

const mockData = {
  photo: {
    id: 1,
    imageUrl: 'https://scontent-ber1-1.cdninstagram.com/vp/88f4d9772eff7bc7e429e3629f507339/5C8BE025/t51.2885-15/e35/44887346_2211003352244412_5881337131845119437_n.jpg',
    description: 'Autumn colours #instagood #followme #instago #fun #my #nice #love #me #cute #follow #photooftheday #like #tbt #beautiful #picoftheday #happy #summer #instadaily #igers #fun #bestoftheday #instamood #instalike #like4like #friends truncated text after three lines ofâ€¦',
    status: 'pending',
    mentioned: true,
    collections: ['cats', 'cute'],
    products: [1],
    source: {
      provider: 'instagram',
      username: 'cats',
      url: 'https://www.instagram.com/p/Bq20cFDnpCB/'
    },
    likes: 9212,
    comments: 82,
    rightsGranted: false,
    sticky: false,
    errors: ['First error', 'Second error']
  },
  selectedPhotos: []
};
const mockhandler = jest.fn();

describe('Thumbnail Component', () => {
  const wrapper = mount(
    <Thumbnail
      photo = {mockData.photo}
      selectedPhotos={mockData.selectedPhotos}
      onStatusChange = {mockhandler}
      onRightsRequest = {mockhandler}
      onStickyChange = {mockhandler}
      onSelectionChange = {mockhandler}
    />
  );

  it('allows us to set props', () => {
    expect(wrapper.props().photo).toBe(mockData.photo);
    expect(wrapper.props().selectedPhotos).toBe(mockData.selectedPhotos);
  });

  wrapper.setState({
    isOver: false
  });

  it('Show overlay binding', () => {
    wrapper.find('div.photo-grid-thumbnail-container').simulate('mouseenter');
    expect(wrapper.state().isOver).toBe(true);
    wrapper.find('div.photo-grid-thumbnail-container').simulate('mouseleave');
    expect(wrapper.state().isOver).toBe(false);
  });
});

describe('Thumbnail SnapShot', () => {
  const wrapper = shallow(
    <Thumbnail
      photo = {mockData.photo}
      selectedPhotos={mockData.selectedPhotos}
      onStatusChange = {jest.fn()}
      onRightsRequest = {jest.fn()}
      onStickyChange = {jest.fn()}
      onSelectionChange = {jest.fn()}
    />
  );

  it('match to snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});