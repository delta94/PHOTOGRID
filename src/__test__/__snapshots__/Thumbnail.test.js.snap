// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnail SnapShot match to snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Thumbnail
    onRightsRequest={[MockFunction]}
    onSelectionChange={[MockFunction]}
    onStatusChange={[MockFunction]}
    onStickyChange={[MockFunction]}
    photo={
      Object {
        "collections": Array [
          "cats",
          "cute",
        ],
        "comments": 82,
        "description": "Autumn colours #instagood #followme #instago #fun #my #nice #love #me #cute #follow #photooftheday #like #tbt #beautiful #picoftheday #happy #summer #instadaily #igers #fun #bestoftheday #instamood #instalike #like4like #friends truncated text after three lines of…",
        "errors": Array [
          "First error",
          "Second error",
        ],
        "id": 1,
        "imageUrl": "https://scontent-ber1-1.cdninstagram.com/vp/88f4d9772eff7bc7e429e3629f507339/5C8BE025/t51.2885-15/e35/44887346_2211003352244412_5881337131845119437_n.jpg",
        "likes": 9212,
        "mentioned": true,
        "products": Array [
          1,
        ],
        "rightsGranted": false,
        "source": Object {
          "provider": "instagram",
          "url": "https://www.instagram.com/p/Bq20cFDnpCB/",
          "username": "cats",
        },
        "status": "pending",
        "sticky": false,
      }
    }
    selectedPhotos={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <StatusButtonGroup
          photo={
            Object {
              "collections": Array [
                "cats",
                "cute",
              ],
              "comments": 82,
              "description": "Autumn colours #instagood #followme #instago #fun #my #nice #love #me #cute #follow #photooftheday #like #tbt #beautiful #picoftheday #happy #summer #instadaily #igers #fun #bestoftheday #instamood #instalike #like4like #friends truncated text after three lines of…",
              "errors": Array [
                "First error",
                "Second error",
              ],
              "id": 1,
              "imageUrl": "https://scontent-ber1-1.cdninstagram.com/vp/88f4d9772eff7bc7e429e3629f507339/5C8BE025/t51.2885-15/e35/44887346_2211003352244412_5881337131845119437_n.jpg",
              "likes": 9212,
              "mentioned": true,
              "products": Array [
                1,
              ],
              "rightsGranted": false,
              "source": Object {
                "provider": "instagram",
                "url": "https://www.instagram.com/p/Bq20cFDnpCB/",
                "username": "cats",
              },
              "status": "pending",
              "sticky": false,
            }
          }
        />,
        <div
          className="triangle"
        >
          <span
            className="icon__star triangle__star font-14"
          />
        </div>,
        false,
      ],
      "className": "photo-grid-thumbnail-container br__radius-5",
      "onMouseEnter": [Function],
      "onMouseLeave": [Function],
      "style": Object {
        "backgroundImage": "url(https://scontent-ber1-1.cdninstagram.com/vp/88f4d9772eff7bc7e429e3629f507339/5C8BE025/t51.2885-15/e35/44887346_2211003352244412_5881337131845119437_n.jpg)",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "photo": Object {
            "collections": Array [
              "cats",
              "cute",
            ],
            "comments": 82,
            "description": "Autumn colours #instagood #followme #instago #fun #my #nice #love #me #cute #follow #photooftheday #like #tbt #beautiful #picoftheday #happy #summer #instadaily #igers #fun #bestoftheday #instamood #instalike #like4like #friends truncated text after three lines of…",
            "errors": Array [
              "First error",
              "Second error",
            ],
            "id": 1,
            "imageUrl": "https://scontent-ber1-1.cdninstagram.com/vp/88f4d9772eff7bc7e429e3629f507339/5C8BE025/t51.2885-15/e35/44887346_2211003352244412_5881337131845119437_n.jpg",
            "likes": 9212,
            "mentioned": true,
            "products": Array [
              1,
            ],
            "rightsGranted": false,
            "source": Object {
              "provider": "instagram",
              "url": "https://www.instagram.com/p/Bq20cFDnpCB/",
              "username": "cats",
            },
            "status": "pending",
            "sticky": false,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span
            className="icon__star triangle__star font-14"
          />,
          "className": "triangle",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "icon__star triangle__star font-14",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        "type": "div",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <StatusButtonGroup
            photo={
              Object {
                "collections": Array [
                  "cats",
                  "cute",
                ],
                "comments": 82,
                "description": "Autumn colours #instagood #followme #instago #fun #my #nice #love #me #cute #follow #photooftheday #like #tbt #beautiful #picoftheday #happy #summer #instadaily #igers #fun #bestoftheday #instamood #instalike #like4like #friends truncated text after three lines of…",
                "errors": Array [
                  "First error",
                  "Second error",
                ],
                "id": 1,
                "imageUrl": "https://scontent-ber1-1.cdninstagram.com/vp/88f4d9772eff7bc7e429e3629f507339/5C8BE025/t51.2885-15/e35/44887346_2211003352244412_5881337131845119437_n.jpg",
                "likes": 9212,
                "mentioned": true,
                "products": Array [
                  1,
                ],
                "rightsGranted": false,
                "source": Object {
                  "provider": "instagram",
                  "url": "https://www.instagram.com/p/Bq20cFDnpCB/",
                  "username": "cats",
                },
                "status": "pending",
                "sticky": false,
              }
            }
          />,
          <div
            className="triangle"
          >
            <span
              className="icon__star triangle__star font-14"
            />
          </div>,
          false,
        ],
        "className": "photo-grid-thumbnail-container br__radius-5",
        "onMouseEnter": [Function],
        "onMouseLeave": [Function],
        "style": Object {
          "backgroundImage": "url(https://scontent-ber1-1.cdninstagram.com/vp/88f4d9772eff7bc7e429e3629f507339/5C8BE025/t51.2885-15/e35/44887346_2211003352244412_5881337131845119437_n.jpg)",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "photo": Object {
              "collections": Array [
                "cats",
                "cute",
              ],
              "comments": 82,
              "description": "Autumn colours #instagood #followme #instago #fun #my #nice #love #me #cute #follow #photooftheday #like #tbt #beautiful #picoftheday #happy #summer #instadaily #igers #fun #bestoftheday #instamood #instalike #like4like #friends truncated text after three lines of…",
              "errors": Array [
                "First error",
                "Second error",
              ],
              "id": 1,
              "imageUrl": "https://scontent-ber1-1.cdninstagram.com/vp/88f4d9772eff7bc7e429e3629f507339/5C8BE025/t51.2885-15/e35/44887346_2211003352244412_5881337131845119437_n.jpg",
              "likes": 9212,
              "mentioned": true,
              "products": Array [
                1,
              ],
              "rightsGranted": false,
              "source": Object {
                "provider": "instagram",
                "url": "https://www.instagram.com/p/Bq20cFDnpCB/",
                "username": "cats",
              },
              "status": "pending",
              "sticky": false,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="icon__star triangle__star font-14"
            />,
            "className": "triangle",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "icon__star triangle__star font-14",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
